src/input.c

```diff
@@ -40,93 +40,173 @@
  */
 typedef struct Keybind_ {
    int disabled; /**< Whether or not it's disabled. */
-   const char *name; /**< keybinding name, taken from keybind_info */
+   const char *brief, detailed; /** */
    KeybindType type; /**< type, defined in player.h */
    SDL_Keycode key; /**< key/axis/button event number */
    SDL_Keymod mod; /**< Key modifiers (where applicable). */
 } Keybind;
 
-/* name of each keybinding */
-const char *keybind_info[][3] = {
+/**
+ * @brief Naev internal key types
+ */
+typedef enum KeySemanticType_ {
+   KST_ACCEL=0,
+   KST_LEFT,
+   KST_RIGHT,
+   KST_REVERSE,
+   KST_FACE,
+
+   KST_STEALTH,
+   KST_GAME_SPEED,
+   KST_PAUSE,
+
+   KST_AUTONAV,
+   KST_APPROACH,
+   KST_MOUSE_FLYING,
+   KST_JUMP
+
+   KST_TARGET_NEXT,
+   KST_TARGET_PREV,
+   KST_TARGET_CLOSE,
+   KST_TARGET_SPOB,
+   KST_TARGET_JUMP,
+
+   KST_HTARGET_NEXT,
+   KST_HTARGET_PREV,
+   KST_HTARGET_CLOSE,
+
+   KST_TARGET_CLEAR,
+
+   KST_FIRE_PRIMARY,
+   KST_FIRE_SECONDARY,
+   KST_INIT_COOLDOWN,
+
+   KST_TAB_1,
+   KST_TAB_2,
+   KST_TAB_3,
+   KST_TAB_4,
+   KST_TAB_5,
+   KST_TAB_6,
+   KST_TAB_7,
+   KST_TAB_8,
+   KST_TAB_9,
+   KST_TAB_0,
+
+   KST_LOCAL_MAP,
+   KST_GLOBAL_MAP,
+
+   KST_MENU_SMALL,
+   KST_MENU_INFO,
+   KST_MENU_LUA,
+
+   KST_ESCORT_ATTACK,
+   KST_ESCORT_HALT,
+   KST_ESCORT_RETURN,
+   KST_ESCORT_CLEAR
+
+   KST_COMM_HAIL,
+   KST_COMM_RECEIVE,
+   KST_COMM_UP,
+   KST_COMM_DOWN
+
+   KST_ZOOM_IN,
+   KST_ZOOM_OUT,
+
+   KST_FULLSCREEN,
+
+   KST_SCREENSHOT,
+
+   KST_PASTE,
+
+} KeySemanticType;
+
+
+/* Description of each key semantic type */
+const char *keybind_info[KST_PASTE+1][2] = {
    /* Movement */
-   { "accel", N_("Accelerate"), N_("Makes your ship accelerate forward.") },
-   { "left", N_("Turn Left"), N_("Makes your ship turn left.") },
-   { "right", N_("Turn Right"), N_("Makes your ship turn right.") },
-   { "reverse", N_("Reverse"), N_("Makes your ship face the direction you're moving from. Useful for braking.") },
-   { "stealth", N_("Stealth"), N_("Tries to enter stealth mode.") },
+   { N_("Accelerate"), N_("Makes your ship accelerate forward.") },
```

Comment:
Would be nicer to probably use some #define magic and explicitly include the KST_ code instead of depending on the order of the enum, which will likely cause issues sometime down the line.

Reply:
Yeah, will fix later on in the draft; prototype bind_desc(enum, short_string, long_string)

Reply:
Bit of a time shortage, so started draft with partially complete code. Will first get this thing working as expected, and then make sanity changes.

Reply:
OK, sounds good. Thanks!